src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -218,13 +218,117 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                         }
                         else
                         {
+#if UNIX
+                            // On UNIX systems, we expand arguments containing wildcard expressions against
+                            // the file system just like bash, etc.
```

Comment:
Looking at the bash source, a lot of the subtleties around globbing are in bash itself, not the glob routine.  For example ~ expansion is not handled by glob. Likewise quote processing is handled in bash itself, not by the glob routine. Given that PowerShell does it's quote removal (and addition) in a very different way than bash, there are going to be differences. (To make /bin/echo "*" work, we need to be able to be able to determine if a string value was parsed as a bareword literal at execution time.) Also note that Bash optionally supports extended globbing  (See [Bash Extended Globbing](http://www.linuxjournal.com/content/bash-extended-globbing) for an overview). Anyway, I've opened a new issue to track this investigation [#3655](https://github.com/PowerShell/PowerShell/issues/3655)